version: "3.8"
services:
 flask:
  container_name: eaton-flask
  build:
   context: ./
   dockerfile: Dockerfile.server
  command: python app.py
  ports:
   - 5000:5000
  networks:
   - server-network
  restart: on-failure:3
api-serveo:
    container_name: app-api-serveo
    build:
      context: ./
      dockerfile: Dockerfile.api
    tty: true
    stdin_open: true
    command: >
      autossh -M 0
      -o ServerAliveInterval=60
      -o ServerAliveCountMax=3
      -o ExitOnForwardFailure=yes
      -o StrictHostKeyChecking=no
      -R dev-app-server:80:app-django:8000
      serveo.net
    networks:
      - server-network
    restart: always

 api-serveo:
  container_name: eaton-api-serveo
  build:
   context: ./
   dockerfile: Dockerfile.serveo
  command: python run.py
  networks:
   - server-network
  restart: on-failure:3

networks:
 server-network:
  driver: bridge
